workflow_obs.py

```diff
@@ -297,24 +317,118 @@ def clean_for_zarr(ds: xr.Dataset) -> xr.Dataset:
                             )
                             ds_output = da_output.to_dataset(name=performance_variable_name) # The output dataset
                             
-                            ds_output.attrs["cat:xrfreq"]= "fx" # Frequency is fixed, as there is no time axis
+                            ds_output.attrs["cat:xrfreq"] = "fx"
                             ds_output.attrs["cat:variable"] = performance_variable_name
                             ds_output.attrs["cat:processing_level"] = "performance"
                             ds_output.attrs["cat:source"] = rec_source
                             ds_output.attrs["cat:performance_base"] = obs_dataset.attrs["cat:id"]
                             ds_output.attrs["cat:domain"] = rec_dataset.attrs["cat:domain"]
                             ds_output.attrs["cat:id"] = xs.catalog.generate_id(ds_output, id_columns=xs.catalog.ID_COLUMNS + ['performance_base'])[0]
```

Comment:
if you regenerate the id here, the check of the catalog (l.261) will not work. The idea of `pcat.exists_in_cat` is to not recalculate things that exists. Right now, it will. I also notice that you are missing variable=performance_variable_name for it to work properly.